### ISSUE TITLE
Inconsistent use of elementwise and non-elementwise operators for multiplication and division

### ISSUE THREAD

**Issue by rok-cesnovar** [stan-dev/stanc3#789]
There are a few inconsistencies with regards to what we allow with elementwise division and multiplication and the non-elementwise division and multiplication operator.

With
```stan 
matrix[5,5] a;
matrix[5,5] c;
real b;
```

This compiles with elementwise division
```stan
c = a ./ b;
c = b ./ a;
c = elt_divide(a, b);
c = elt_divide(b, a);
```
but the same signatures do not work for elementwise multiplication.
```stan
c = a .* b;
c = b .* a;
c = elt_multiply(a, b);
c = elt_multiply(b, a);
```

There is a vice-versa situation with the non-elementwise operator. This compiles:
```stan
c = a * b;
c = b * a;
c = multiply(a, b);
c = multiply(b, a);

c = divide(a, b);
c = a / b;
```
but this does not
```stan
c = divide(b, a);
c = b / a;
```

**Comment by adamhaber** [comment]
I can try to tackle this. Can you point me to where the problem might be?

**Comment by rok-cesnovar** [comment]
I think the problem is just that the missing signatures are not defined.

See elt_divide/elt_multiply for https://github.com/stan-dev/stanc3/blob/master/src/middle/Stan_math_signatures.ml#L907
And here for divide: https://github.com/stan-dev/stanc3/blob/master/src/middle/Stan_math_signatures.ml#L873

So just adding missing ones and some test models. Manual labor :) 

I think some of these are missing Stan Math support though, but I can add that in the meantime.

**Comment by adamhaber** [comment]
No problem, will try to finish this in the next couple of days

**Comment by adamhaber** [comment]
Closed by #791

**Comment by WardBrian** [comment]
Since #791 got reverted by #793 I'm going to reopen this
